torch/distributions/bernoulli.py

```diff
@@ -33,14 +33,16 @@ def __init__(self, probs=None, logits=None):
         if (probs is None) == (logits is None):
             raise ValueError("Either `probs` or `logits` must be specified, but not both.")
         if probs is not None:
+            is_scalar = isinstance(probs, Number)
             self.probs, = broadcast_all(probs)
         else:
+            is_scalar = isinstance(logits, Number)
             self.logits, = broadcast_all(logits)
-        probs_or_logits = probs if probs is not None else logits
-        if isinstance(probs_or_logits, Number):
+        self.probs_or_logits = self.probs if probs is not None else self.logits
```

Comment:
:+1: to adding this field. Consider renaming to `self._prototype` or `self._template` or `self._param`?

Reply:
Sure, will rename to `self._param`. I get warnings that a private property is accessed from outside the class, but it should be fine since this isn't part of the API and only used internally within distributions.

Reply:
Oh interesting, is that in your IDE? I guess those meaning of underscore both make sense (private to class -vs- not part of public API), and I hadn't thought of that before. In this case it seems cleaner to ensure it's not part of the public API.

What might make sense to add to the public API would be a `.new()` method and a `.param_shape` property (as you have added in this PR). Are there any attributes we use of `_param` other than `.shape` and `.new()`?

Reply:
Adam said adding a `.new()` method is pretty common practice, so `.new()` + `.param_shape` (both public) seems pretty clean. WDYT?

Reply:
If we add a `.new()` method here, we could send another PR that adds them in other distributions and simplifies a bunch of the `_kl_infinite` logic.